simde/x86/sse4.2.h

```diff
@@ -37,6 +37,231 @@ SIMDE__BEGIN_DECLS
+#define SIMDE_SIDD_CMP_RANGES     1
+#define SIMDE_SIDD_CMP_EQUAL_EACH     2
+#define SIMDE_SIDD_CMP_EQUAL_ORDERED   3
+
+#if defined(SIMDE_X86_SSE4_2_ENABLE_NATIVE_ALIASES)
+#define _SIDD_CMP_EQUAL_ANY SIMDE_SIDD_CMP_EQUAL_ANY
+#define _SIDD_CMP_RANGES SIMDE_SIDD_CMP_RANGES
+#define _SIDD_CMP_EQUAL_EACH SIMDE_SIDD_CMP_EQUAL_EACH
+#define _SIDD_CMP_EQUAL_ORDERED SIMDE_SIDD_CMP_EQUAL_ORDERED
+#endif
+
+SIMDE__FUNCTION_ATTRIBUTES
+int
+simde_mm_cmpestra_8_(simde__m128i a, int la, simde__m128i b, int lb, const int imm8) {
+  const int cmp_op = imm8 & 0x06;
+  const int polarity = imm8 & 0x30;
+  simde__m128i_private
+    bool_res_ = simde__m128i_to_private(simde_mm_setzero_si128()),
+    a_ = simde__m128i_to_private(a),
+    b_ = simde__m128i_to_private(b);
+  const int upper_bound = (128 / 8) - 1;
+  int a_invalid = 0;
+  int b_invalid = 0;
+  for(int i = 0 ; i < (upper_bound) ; i++) {
+    for(int j = 0; j< (upper_bound) ; j++){
+      int bitvalue = ((a_.i8[i] == b_.i8[j]) ? 1 : 0);
+      bool_res_.i8[i] |= (( bitvalue ) << j);
+      if(i == la)
+        a_invalid = 1;
+      if(j == lb)
+        b_invalid = 1;
+      switch(cmp_op){
+        case SIMDE_SIDD_CMP_EQUAL_ANY:
+          break;
+        case SIMDE_SIDD_CMP_RANGES:
+          break;
+        case SIMDE_SIDD_CMP_EQUAL_EACH:
+          if(a_invalid && b_invalid)
+            bool_res_.i8[i] |= (1 << j);
+          break;
+        case SIMDE_SIDD_CMP_EQUAL_ORDERED:
+          if(a_invalid && !b_invalid)
+            bool_res_.i8[i] |= (1 << j);
+          else if(a_invalid && b_invalid)
+            bool_res_.i8[i] |= (1 << j);
+          break;
+      }
+    }
+  }
+  int32_t int_res_1 = 0;
+  int32_t int_res_2 = 0;
+  switch(cmp_op) {
+    case SIMDE_SIDD_CMP_EQUAL_ANY:
+      for(int i = 0 ; i < (upper_bound) ; i++){
+        SIMDE__VECTORIZE_REDUCTION(|:int_res_1)
+        for(int j = 0 ; j < (upper_bound) ; j++){
+          int_res_1 |= (((bool_res_.i8[i] >> j) & 1) << i);
+        }
+      }
+      break;
+    case SIMDE_SIDD_CMP_RANGES:
+      for(int i = 0 ; i < (upper_bound) ; i++){
+        SIMDE__VECTORIZE_REDUCTION(|:int_res_1)
+        for(int j = 0 ; j < (upper_bound) ; j++){
+          int_res_1 |= ((((bool_res_.i8[i] >> j) & 1) & ((bool_res_.i8[i] >> (j + 1)) & 1)) << i);
+          j += 2;
+        }
+      }
+      break;
+    case SIMDE_SIDD_CMP_EQUAL_EACH:
+      for(int i = 0 ; i < (upper_bound) ; i++){
+        SIMDE__VECTORIZE_REDUCTION(|:int_res_1)
+        for(int j = 0 ; j < (upper_bound) ; j++){
+          int_res_1 |= (((bool_res_.i8[i] >> i) & 1) << i);
+        }
+      }
+      break;
+    case SIMDE_SIDD_CMP_EQUAL_ORDERED:
+      int_res_1 = (imm8 & 1) ? 0xff : 0xffff;
+      for(int i = 0 ; i < (upper_bound) ; i++){
+        int k = i;
+        SIMDE__VECTORIZE_REDUCTION(|:int_res_1)
+        for(int j = 0 ; j < (upper_bound-i) ; j++){
+          int_res_1 &=  (((bool_res_.i8[k] >> j) & 1 ) << i) ;
+          k += 1;
+        }
+      }
```

Comment:
corrected this, I thought as this was last case so it was ok to omit it

Reply:
Technically I think it is; it's possible some compilers will warn about falling through, I'm honestly not sure.  However, I strongly value consistency in code.  Having a break at the end means there is no special case, which I think makes it easier to understand.

You may think I'm being pedantic, and maybe I am, but C is a dangerous language.  You have to do everything you can to make prevent bugs.  Part of that is giving the compiler as much information as you can so it can give you good diagnostics (I hope you're compiling with at least -Wall), but part of it is also making sure the code is as easy to read as possible.  Making it easier to read makes it easier to review, which increases the odds of finding any bugs in the code.